### DISCUSSION TITLE
How to retain a command from a hard-wired button

### DISCUSSION THREAD

**Discussion by pvoncken** [dawidchyrzynski/arduino-home-assistant#233] [Q&A]
I have a few switches that work perfect in HA.
But now i also added local hardwired buttons to switch the lights manually.
It works fine except that state changes that are made locally are not retained.  Only the state-changes made in HA are retained.
Does anyone have an idea to make this also work for the local changes?

**Comment by jwilleke** [comment]
I have a working "test" sample that might be helpful.

- that has two-way communication from "switches" in home assistant to Arduino
- a little complex as I have several controlled items
- always a work in progress 
 
at  [ArduinoHA-examples](https://github.com/jwilleke/ArduinoHA-examples/blob/main/src/main.cpp)

Questions, Comments and Suggestions for improvement are always encouraged!

**Reply by pvoncken** [reply]
Thanxs,
I'll take a look at it.

**Reply by pvoncken** [reply]
I solved it by using the mqtt.publish function and sending an MQTT ON/OFF command message when de local buttons are pressed.
In the previous version when a button was pressed i reported the state change back to HA.
Apparently a state change is not retained bij the MQTT server.
